Lib/test/test_capi/test_opt.py

```diff
@@ -1603,6 +1603,32 @@ def testfunc(n):
         self.assertNotIn("_COMPARE_OP_INT", uops)
         self.assertIn("_POP_TWO_LOAD_CONST_INLINE_BORROW", uops)
 
+    def test_to_bool_bool_contains_op_set(self):
+        """
+        Test that _TO_BOOL_BOOL is removed from code like:
+
+        res = foo in some_set
+        if res:
+            ....
+
+        """
+        def testfunc(n):
+            x = 0
+            s = {1, 2, 3}
+            for _ in range(n):
+                a = 2
+                in_set = a in s
+                if in_set:
+                    x += 1
```

Comment:
We might be able to simplify this further,  I wasn't sure how much the peepholer would optimize out if I used e.g. `2 in s` directly.

Reply:
`2 in s` would have been fine, but this works well too!